service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -351,13 +363,33 @@ public void deleteSite(JWTData userData, int siteId) {
     site.store();
   }
 
-  public void deleteStewardship(JWTData userData, int activityId) {
+  @Override
+  public void editStewardship(JWTData userData, int activityId, EditStewardshipRequest editStewardshipRequest) {
+    checkStewardshipExists(activityId);
     StewardshipRecord activity =
         db.selectFrom(STEWARDSHIP).where(STEWARDSHIP.ID.eq(activityId)).fetchOne();
 
-    if (activity == null) {
-      throw new ResourceDoesNotExistException(activityId, "Stewardship Activity");
+    if (!(activity.getUserId().equals(userData.getUserId())
+        || userData.getPrivilegeLevel().equals(PrivilegeLevel.SUPER_ADMIN)
+        || userData.getPrivilegeLevel().equals(PrivilegeLevel.ADMIN))) {
+      throw new AuthException(
+          "User needs to be an admin or the activity's author to edit the record.");
     }
```

Comment:
🔧 Since this logic is used twice, can we abstract this out? Also, the `isAdmin(PrivilegeLevel)` method (on line 107) can be used to simplify the logic that checks if the user is an admin or super admin.

Reply:
^^